File: CSI2120/assignment/comprehensive/part2/src/employer/employer.go
package employer

import "person"

// Employer is a struct to represent employers
type Employer struct {
	P   *person.Person
	Cur int64
}

// NewEmployer is the constructor of Employer struct
func NewEmployer(p *person.Person) *Employer {
	return &Employer{p, 0}
}

func (e *Employer) getCurrent() string {
	name := e.P.Pref[e.Cur]
	e.Cur++
	return name
}

File: CSI2120/assignment/comprehensive/part1/Employer.java
/**
 * Employer 
 * Hongyi Lin, 300053082
 */
public class Employer extends Person {
    // attribute
    private int current; // current index of the pref array
    // constructor
    public Employer(String name, int n) {
        super(name, n);
        current = 0;
    }
    public Employer(String name, String[] pref) {
        super(name, pref);
        current = 0;
    }
    // method
    /*
     * return the name of the current choice of the student and increment the current
     */
    public String getCurrent() {
        return pref[current++];
    }
}